cartographer/mapping_3d/imu_integration.h

```diff
@@ -36,25 +36,19 @@ struct IntegrateImuResult {
   Eigen::Quaternion<T> delta_rotation;
 };
 
-// Returns velocity delta in map frame.
-IntegrateImuResult<double> IntegrateImu(
-    const std::deque<sensor::ImuData>& imu_data, const common::Time start_time,
-    const common::Time end_time,
-    std::deque<sensor::ImuData>::const_iterator* it);
-
-template <typename T>
+template <typename T, typename RangeType, typename IteratorType>
```

Comment:
I find
T -> NumberType
RangeType -> ContainerType
easier to read.

Reply:
Not feeling strongly but `T` is here to support Ceres jets, i.e. not just a number but including derivative information. This is consistent with `common/math.h` and `transform/transform.h`. I'm open to better suggestions.

`RangeType` is chosen because in the follow up this will not be a container.